modules/core/src/encoder/SemanticPreEncoder.ts

```diff
@@ -61,9 +61,17 @@ export class SemanticPreEncoder {
                 /**
                  * Per June 2021 Policy change, Vendors declaring only Special Purposes must
                  * have their legitimate interest Vendor bit set if they have been disclosed.
-                 * This empty block ensures their LI bit remains set
+                 * This block ensures their LI bit remains set
                  */
-
+                vector.set(vendorId);
+                } else if (gvlVendorKey === 'legIntPurposes' && vendor['purposes'].length > 0 && vendor['legIntPurposes'].length === 0 && vendor['specialPurposes'].length > 0
```

Comment:
@HeinzBaumann I'm confused by this change. Isn't this essentially the equivalent of removing `vendor['purposes'].length === 0` above? It doesn't seem like the `else` is necessary if that change were made.

Additionally, I'm curious to see an example of a CMP using the special purposes. Do we just pass in all of the publisher's property's vendors and let this code do the work of identifying the appropriate list?